In [4]:
# Loading Data
import pandas as pd
xls=pd.ExcelFile("AWS_Sales.xlsx")
xls.sheet_names


['aws_sales',
 'customers',
 'calendar',
 'territories',
 'products_cat',
 'products_subcat',
 'products',
 'returns']

In [5]:
# Data Inventory
for sheet in xls.sheet_names:
    df=pd.read_excel(xls,sheet_name=sheet)
    print(f"{sheet}:{df.shape}")
    

aws_sales:(56046, 13)
customers:(18148, 21)
calendar:(912, 25)
territories:(10, 4)
products_cat:(4, 2)
products_subcat:(37, 3)
products:(293, 14)
returns:(1809, 4)


| Table           |   Rows | Columns | Role           |
| --------------- | -----: | ------: | -------------- |
| aws_sales       | 56,046 |      13 | Fact Table     |
| customers       | 18,148 |      21 | Dimension      |
| calendar        |    912 |      25 | Dimension      |
| territories     |     10 |       4 | Dimension      |
| products_cat    |      4 |       2 | Dimension      |
| products_subcat |     37 |       3 | Dimension      |
| products        |    293 |      14 | Dimension      |
| returns         |  1,809 |       4 | Secondary Fact |


In [29]:
aws_sales=pd.read_excel("AWS_Sales.xlsx",
    sheet_name="aws_sales")

customers = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="customers"
)
calendar = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="calendar"
)
territories = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="territories"
)
products_cat = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products_cat"
)
products_subcat = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products_subcat"
)
products = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="products"
)
returns = pd.read_excel(
    "AWS_Sales.xlsx",
    sheet_name="returns"
)


In [18]:
#Primary Keys
customers.columns.tolist

<bound method IndexOpsMixin.tolist of Index(['CustomerKey', 'Prefix', 'FirstName', 'LastName', 'FullName',
       'BirthDate', 'MaritalStatus', 'Gender', 'EmailAddress', 'AnnualIncome',
       'TotalChildren', 'EducationLevel', 'Occupation', 'HomeOwner',
       'UserName', 'DomainName', 'BirthYear_CC', 'CurrentAge', 'BirthMonth',
       'Parent', 'Full_Name_CC'],
      dtype='str')>

In [19]:
products.columns.tolist()

['ProductKey',
 'ProductSubcategoryKey',
 'ProductSKU',
 'ProductName',
 'ModelName',
 'ProductDescription',
 'ProductColor',
 'ProductSize',
 'ProductStyle',
 'ProductCost',
 'ProductPrice',
 'Discounted_Price',
 'Profit_Before_Discount',
 'Profit_After_Discount']

In [20]:
products_subcat.columns.tolist()

['ProductSubcategoryKey', 'SubcategoryName', 'ProductCategoryKey']

In [21]:
products_cat.columns.tolist()

['ProductCategoryKey', 'CategoryName']

In [22]:
territories.columns.tolist()

['SalesTerritoryKey', 'Region', 'Country', 'Continent']

In [30]:
aws_sales.columns.tolist()

['Source.Name',
 'OrderDate',
 'StockDate',
 'OrderNumber',
 'ProductKey',
 'CustomerKey',
 'TerritoryKey',
 'OrderLineItem',
 'OrderQuantity',
 'QuanityType',
 'RelatedPrice',
 'RelatedDayName',
 'Sales']

In [31]:
returns.columns.tolist()

['ReturnDate', 'TerritoryKey', 'ProductKey', 'ReturnQuantity']

In [32]:
calendar.columns.tolist()

['Date',
 'Year',
 'Start of Year',
 'End of Year',
 'Month',
 'Start of Month',
 'End of Month',
 'Days in Month',
 'Month Name',
 'Quarter',
 'Start of Quarter',
 'End of Quarter',
 'Week of Year',
 'Week of Month',
 'Start of Week',
 'End of Week',
 'Day',
 'Day of Week',
 'Day of Year',
 'Start of Day',
 'End of Day',
 'Day Name',
 'DayOfWeek',
 'Weekened',
 'ShortDay']

In [36]:
# Validate Relationship
set(aws_sales['CustomerKey'])-set(customers['CustomerKey'])

set()

In [39]:
set(aws_sales['ProductKey'])-set(products['ProductKey'])

set()

In [40]:
set(aws_sales['TerritoryKey'])-set(territories['SalesTerritoryKey'])


set()

In [ ]:
# Every customerkey in aws_sales exists in customers, same applies to proucts and territories.
#Our data model is clean and ready for analysis

In [45]:
print(aws_sales['CustomerKey'].nunique())
customers['CustomerKey'].nunique()



17416


18148

In [46]:
print(aws_sales['ProductKey'].nunique())
products['ProductKey'].nunique()

130


293

In [47]:
print(aws_sales['TerritoryKey'].nunique())
territories['SalesTerritoryKey'].nunique()

10


10

Key Findings

1.Customer dimension contains 18,148 customers, of which 17,416 generated sales.

2.Product dimension contains 293 products, but only 130 products appear in sales transactions.

3.All 10 territories have recorded sales.

4.Referential integrity checks revealed no missing dimension references.